pkg/proxy/topology.go

```diff
@@ -43,11 +64,15 @@ func FilterTopologyEndpoint(currentNodeName types.NodeName, nodeMap NodeMap, top
 			if nodeName == "" {
 				continue
 			}
-			node, ok := nodeMap[nodeName]
-			if !ok {
+			node, found := nodeMap[nodeName]
+			if !found {
+				continue
+			}
+			targetLabels := node.Labels()
+			if targetLabels == nil {
 				continue
```

Comment:
Here, we have the node and it explicitly has no labels (although that is suspicious), so I would think continue here is OK.